python/ray/dataframe/test/test_dataframe.py

```diff
@@ -954,10 +954,31 @@ def test_as_blocks():
 
 
 def test_as_matrix():
-    ray_df = create_test_dataframe()
+    test_data = TestData()
+    frame = rdf.DataFrame(test_data.frame)
+    mat = frame.as_matrix()
+
+    frameCols = frame.columns
+    for i, row in enumerate(mat):
+        for j, value in enumerate(row):
+            col = frameCols[j]
+            if np.isnan(value):
+                assert np.isnan(frame[col][i])
+            else:
+                assert value == frame[col][i]
 
-    with pytest.raises(NotImplementedError):
-        ray_df.as_matrix()
+    # mixed type
+    mat = rdf.DataFrame(test_data.mixed_frame).as_matrix(['foo', 'A'])
+    assert mat[0, 0] == 'bar'
+
+    df = rdf.DataFrame({'real': [1, 2, 3], 'complex': [1j, 2j, 3j]})
+    mat = df.as_matrix()
+    assert mat[0, 0] == 1j
+
+    # single block corner case
+    mat = rdf.DataFrame(test_data.frame).as_matrix(['A', 'B'])
+    expected = test_data.frame.reindex(columns=['A', 'B']).values
+    tm.assert_almost_equal(mat, expected)
```

Comment:
Can you test array(df) here too.

Reply:
Please clarify what you mean by `array(df)`.

Reply:
Test the `__array__` function here.

Reply:
We already have a `test___array__`, why do we need to test that here also?

Reply:
Sorry, forgot about that. In that case, this is not necessary.